core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -834,6 +839,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
 
         case None => // This is not applied to single event log file.
       }
+    } catch {
+      case e: Exception => logError(s"Exception while compacting log for $rootPath", e)
```

Comment:
`ERROR` level seems to be very high. Can we lower the log level, `WARN`? This is printed at every period, isn't it?

Reply:
That's consistent with existing exception handling for `mergeApplicationListing`.

https://github.com/apache/spark/blob/5916c7d0d0a68266ede1b7e54c3498a18142cf8f/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L654-L690

`compact` is even executed only when `mergeApplicationListing` succeeds, which would prune many of possible error cases for `compact`, so less possible to be logged.

If we feel the log level of both are too high, I'll change both altogether. WDYT?

Reply:
The above error handling is more fine-grained that this.
For example, `AccessControlException` is the one which most users will hit frequently with this feature. (That's the reason why I asked at the documentation.) For `AccessControlException`, the above code shows `logWarning` instead of `Exception`. That seems to be the difference between our perspectives.

Let's see @vanzin or other people's opinion.

Reply:
Ah that's good point. I agree with you that `mergeApplicationListing` has more fine-grained exception handling and I even think `compact` should have same handling except blacklisting. Let me fix it.